docs/pages/_app.js

```diff
@@ -1,19 +1,114 @@
-import React from 'react'
+import React from 'react';
 import * as Sentry from '@sentry/browser';
-import App, { Container } from 'next/app'
+import App, { Container } from 'next/app';
 
-Sentry.init({dsn: "https://67e35a01698649d5aa33aaab61777851@sentry.io/1526800"});
+Sentry.init({
+  dsn: 'https://67e35a01698649d5aa33aaab61777851@sentry.io/1526800',
+  beforeSend(event) {
+    let message = getMessage(event);
 
-class MyApp extends App {
+    // If we don't know about this particular type of event then just pass it along
+    if (!message) {
+      return event;
+    }
+
+    // Discard any errors that we know we do not care about
+    if (ERRORS_TO_DISCARD.includes(message)) {
+      return null;
+    }
+
+    // Only attempt to check against cached reported messages if we have localStorage
+    if (window.localStorage) {
+      // Clear the saved error messages every day
+      maybeResetReportedErrorsCache();
+
+      // Bail out if we have reported the error already
+      if (userHasReportedErrorMessage(message)) {
+        return null;
+      }
+
+      saveReportedErrorMessage(message);
+    }
+
+    return event;
+  },
+});
+
+export default class MyApp extends App {
   render() {
-    const { Component, pageProps } = this.props
+    const { Component, pageProps } = this.props;
 
     return (
       <Container>
         <Component {...pageProps} />
       </Container>
-    )
+    );
   }
 }
 
-export default MyApp
\ No newline at end of file
+/*
```

Comment:
Would be nice to move this into a file like error-logging.js that _app.js imports and initializes